tpgtools/overrides/compute/samples/networkfirewallpolicyrule/global.tf.tmpl

```diff
@@ -0,0 +1,41 @@
+resource "google_compute_network_firewall_policy" "basic_network_firewall_policy" {
+  name = "{{policy}}"
+  project = "{{project}}"
+  description = "Sample global network firewall policy"
+}
+
+resource "google_compute_network_firewall_policy_rule" "primary" {
```

Comment:
Small barrage of questions about the `name` field, sorry about that 🙂 

1. `name` is called `ruleName` in the API, and I'm not sure why it was mapped onto `name`, w/ no context in the original DCL change. We'd prefer consistency w/ the API & once this ships it's stuck whatever we call it indefinitely- can we change it to match? 

1. `name` has the description "An optional description for this resource.". In the API the description is "An optional name for the rule. This field is not a unique identifier and can be updated.", so I suspect that's a copy-paste artifact (of the `description` field description`). We'll want to fix that the the DCL layer, although since it's just a description it's not merge-blocking.

1. `name` doesn't appear in a test, and I'd _prefer_ it appears in one. As a `description`-like field it's pretty low stakes so this is non-blocking.

Reply:
So after trying to update this in DCL I'm stuck between two options, save for modifying the main test generator template in DCL.
Either we keep `name` as `ruleName` like it is here. Or we change it to `rule_name` but then I'll have to update `match.src_secure_tags.name` to `match.src_secure_tags.tag_name` (and same for target_secure_tags).

There is a code block in the test generator template that is inserting a list call with a ref to an `item.Name` field if there is _any_ nested field = name. We're uniquely affected here because of the existence of `secure_tags.name` fields.

Either way we're departing from API nomenclature, or should we keep it as "name" and do a custom override here to map name -> rule_name ?

Reply:
Discussed offline, there's a bug with the DCL here. I (hopefully) have a fix out.

Reply:
thanks! updated in latest commit